guides/source/active_job_basics.md

```diff
@@ -131,77 +119,330 @@ That's it!
+
+Active Job supports positive integer priorities when enqueuing jobs. You can read more about the [Priority section](#priority). This setup is helpful when you have jobs with different levels of importance or urgency in the same queue. Within a single queue, jobs are picked based on their priority (with lower values being higher priority). However, when you have multiple queues, the order of the queues themselves takes priority.
+
+For example, if you have two queues, `production` and `background`, jobs in the `production` queue will always be processed first, even if some jobs in the `background` queue have a higher priority.
+
+WARNING: You should avoid using the `priority` option if you're relying on the
+order of the queues list. If both the order of the queues and the priority
+option are used, the queue order will take precedence, and the priority option
+will only apply within each queue.
+
+### Threads, Processes, and Signals
+
+In Solid Queue, parallelism is achieved through threads (configurable via the [`threads` parameter](#configuration)), processes (via the [`processes` parameter](#configuration)), or horizontal scaling. The supervisor manages processes and responds to the following signals:
+
+- **TERM, INT**: Starts graceful termination, sending a TERM signal and waiting up to `SolidQueue.shutdown_timeout`. If not finished, a QUIT signal forces processes to exit.
+- **QUIT**: Forces immediate termination of processes.
+
+If a worker is killed unexpectedly (e.g., with a `KILL` signal), in-flight jobs are marked as failed, and errors like `SolidQueue::Processes::ProcessExitError` or `SolidQueue::Processes::ProcessPrunedError` are raised. Heartbeat settings help manage and detect expired processes. Read more about [Threads, Processes and Signals in the Solid Queue documentation](https://github.com/rails/solid_queue?tab=readme-ov-file#threads-processes-and-signals).
+
+### Lifecycle Hooks
+
+Solid Queue provides lifecycle hooks that let you run code at specific points in the supervisor’s and worker’s lifecycles. The two supervisor lifecycle hooks are:
+
+- `start`: Runs after the supervisor boots but before it forks workers and dispatchers.
+- `stop`: Runs after receiving a shutdown signal (`TERM`, `INT`, or `QUIT`) but before graceful or immediate shutdown begins.
+
+Similarly, there are worker lifecycle hooks:
+
+- `worker_start`: Runs after a worker boots but before it starts polling.
+- `worker_stop`: Runs after receiving a shutdown signal but before the worker shuts down.
+
+You can use methods like `SolidQueue.on_start` and `SolidQueue.on_worker_start` to hook into these lifecycle points. For example:
+
+```ruby
+SolidQueue.on_start { start_metrics_server }
+SolidQueue.on_stop { stop_metrics_server }
```

Comment:
It seems, anytime we call methods on SolidQueue directly it doesn't belong in this guide?
These APIs can change and are a burden to keep in-sync.

Reply:
That's a good rule of thumb, I'll do that thanks!